src/api/index.ts

```diff
@@ -1,9 +1,58 @@
-import goods from '../assets/goods.json';
-import categories from "../assets/categories.json"
+import { Category, Good } from "../types";
+import { GoodInCart } from "store/cart/slice";
 
-const get = <T>(mock: T): Promise<T> => new Promise((resolve) => setTimeout(() => resolve(mock), 1_000));
+const BASE_URL = "http://localhost:3000/";
 
+const getData = <T = unknown>(url: string, params: Record<string, string | number> = {}): Promise<T> => {
+       const searchParams = new URLSearchParams({ ...params } as Record<string, string>);
 
-export const getGoods = () => get(goods);
+       const fullUrl = new URL(url, BASE_URL);
 
-export const getCategories = () => get(categories);
\ No newline at end of file
+       fullUrl.search = searchParams.toString();
+       return fetch(fullUrl)
+              .then((data) => {
+                     if (data.ok) {
+                            return data.json();
+                     }
+                     throw new Error("network is offline");
+              });
+}
+
+const postData = (url: string, body: Record<string, unknown>) => {
+       const fullUrl = new URL(url, BASE_URL);
+       return fetch(fullUrl, {
+              method: "POST",
+              body: JSON.stringify(body),
+       }).then((data) => {
+              if (data.ok) return data.json();
+       })    
+};
+
+const putInCart = (url:string, body: Record<string, unknown>) => {
```

Comment:
это функция для любого put запроса по аналогии с getData и postData выше. И луше везде поубирать постфикс data из названия т.к. он дополнительной смысловой нагрузки чего не несет и назвать просто get / post / put 
```suggestion
const putData = (url:string, body: Record<string, unknown>) => {
```